### PR TITLE
Rename Bumped to CollidedWith

### PR DESCRIPTION
**Pull request by optimumtact**
In light of the change of Bump to Collide, this should also be done
as it will make it less confusing for devs in the future

I also ended up changing a bunch of the code in the CollidedWith procs,
things that were assuming types, doing bad loc changes and so forth

### CODE REVIEW

File: code/game/machinery/recycler.dm [hunk 1]
```diff
@@ -92,10 +92,10 @@
 /obj/machinery/recycler/Bump(atom/movable/AM)
```
Comment (optimumtact) [inline code/game/machinery/recycler.dm]: This whole proc can probably be deleted, it's nto doing anything it's parent procs arent

File: code/game/objects/effects/anomalies.dm [hunk 2]
```diff
@@ -146,8 +147,9 @@
 /obj/effect/anomaly/flux/Bump(mob/living/M)
 	mobShock(M)
 
-/obj/effect/anomaly/flux/Bumped(mob/living/M)
-	mobShock(M)
+/obj/effect/anomaly/flux/CollidedWith(atom/movable/AM)
+	if(isliving(AM))
```
Comment (AnturK) [inline code/game/objects/effects/anomalies.dm]: This and above isliving check is already done in the inner function.

File: code/game/objects/structures/grille.dm [hunk 3]
```diff
@@ -49,17 +49,19 @@
...
-				tile_density = TRUE
-				break
-		if(!tile_density)
-			shock(user, 70)
+/obj/structure/grille/CollidedWith(atom/movable/AM)
+	if(!ismob(AM))
+		return
+	var/mob/M = AM
+	var/tile_density = FALSE
+	for(var/atom/movable/TAM in get_turf(src))
```
Comment (AnturK) [inline code/game/objects/structures/grille.dm]: This one is likely unnecessary this is handled in core Collide code now.

File: code/game/objects/effects/anomalies.dm (hunk 2 above)
Comment (optimumtact) [inline code/game/objects/effects/anomalies.dm]: yah but that's gross

File: code/game/objects/structures/grille.dm (hunk 3 above)
Comment (optimumtact) [inline code/game/objects/structures/grille.dm]: figures, kk

Comment (MrStonedOne) [inline code/game/objects/structures/grille.dm]: So this is 1: double bumping, and two, also calling on bumped(), so thats 3 times grills shock you?

Am I understanding that right?

wow,

Comment (optimumtact) [inline code/game/objects/structures/grille.dm]: I think they're just talking about the layers

Comment (AnturK) [inline code/game/objects/structures/grille.dm]: Yeah, just the layer part, before core bump just used contents order do decide what gets bumped first, now it's done from top in all cases.

File: code/game/machinery/recycler.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/machinery/recycler.dm]: I did this in my original bump rename proc, so this is dealt with

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**